fix: Remove doubled occurrence of periods in simulation

pjaspers opened the pull request:
This was triggered because the backend just returns all the org_units
it discovered in the simulation, not just the selected one.

Since fixing this at the backend layer would be a lot of strain on the
Rails server (it would need to pull down the JSON blob from S3, parse
it, select only the correct parts, then push the JSON blob back to the
frontedn), for now we fix it on the front end.

On the plus side, this makes the simulation page faster because it has
less to render.

Commit:
fix: Remove doubled occurrence of periods in simulation

This was triggered because the backend just returns all the org_units
it discovered in the simulation, not just the selected one.

Since fixing this at the backend layer would be a lot of strain on the
Rails server (it would need to pull down the JSON blob from S3, parse
it, select only the correct parts, then push the JSON blob back to the
frontedn), for now we fix it on the front end.

On the plus side, this makes the simulation page faster because it has
less to render.

borisrorsvort commented on src/components/Simulation/SimulationBlocks.js:
```diff
@@ -20,7 +20,16 @@ const SimulationBlocks = props => {
   const [error, setError] = useState(null);
   const [loading, setLoading] = useState(false);
-  const setsByCode = groupBy((data || {}).invoices, "code");
+  const invoices = (data || {}).invoices || [];
```
const naming should not refer to invoices but sets.

borisrorsvort commented on src/components/Simulation/SimulationBlocks.js:
```diff
@@ -20,7 +20,16 @@ const SimulationBlocks = props => {
+  // Placeholder to only display the selected org unit, in the future
+  // the backend should do this filtering for us.
+  const displayedOrgUnit = props.searchQuery.orgUnit || "";
+  const invoicesForOrgUnit = invoices.filter(
```
Please rebase from develop as optimized the filtering perf. This should be probably integrated within the memoized function, otherwise it will re-filter on each re-render. See https://github.com/BLSQ/hesabu-manager/commit/a1d80f29780b91cc8ccf1fc82cc66b364d94892e#diff-124b0b37de1543bcd7c6e2e1e2d9ca00R61

Commit:
Memoize and rename variable

Now the sets will only be filtered ones, I've also removed the
`invoices` wording since we're now talking about sets.
